app/models/invitation.rb

```diff
@@ -60,12 +60,18 @@ def sent?
   def notify_invitee
     # TODO: maybe subclass to Invitation and ProviderInvitation
     if account.provider?
-      ProviderInvitationMailer.invitation(self).deliver_now
+      ProviderInvitationMailer.register_observer(InvitationDeliveryObserver)
+      ProviderInvitationMailer.invitation(self).deliver_later
     else
-      InvitationMailer.invitation(self).deliver_now
+      InvitationMailer.register_observer(InvitationDeliveryObserver)
+      InvitationMailer.invitation(self).deliver_later
     end
+  end
```

Comment:
We should DRY this. Suggestion below 👇 
We should also ensure that all this has a test.

```ruby
def notify_invitee
  mailer_klass = account.provider? ? ProviderInvitationMailer : InvitationMailer
  mailer_klass.register_observer(InvitationDeliveryObserver)
  mailer_klass.invitation(self).deliver_later
end
```

Reply:
Sounds good. I think the observer is going to be "permanent" tho. It's supposed to be added to Mailer::Base once at initialization time.. but this would mean the PR will be much bigger than we expected, how strange 😉